### PR TITLE
Added a parameter to output body to curldbg option

### PR DESCRIPTION
**Pull request by ggtakec**
### Relevant Issue (if applicable)
#1289 

### Details
Added a parameter to output `body` to curldbg option.

The curldbg option can be specified by `curldbg=normal`, ` curldbg=body` or the same `curldbg` as before.
If the parameter is omitted, it is the same as `normal`.

`curldbg` or` curldbg=normal` is the same debug output as before.
When `curldbg=body` is specified, in addition to this debug output, the request body of `CompleteMultipartUpload` and the response body of `ListObjects` are output.

In this PR, output body is limited to a specific API.
In the future, if necessary, we will add APIs or change the APIs to output body output.

### CODE REVIEW

File: src/curl.cpp [hunk 1]
```diff
@@ -3529,6 +3580,9 @@ int S3fsCurl::ListBucketRequest(const char* tpath, const char* query)
   curl_easy_setopt(hCurl, CURLOPT_URL, url.c_str());
   curl_easy_setopt(hCurl, CURLOPT_WRITEDATA, (void*)&bodydata);
   curl_easy_setopt(hCurl, CURLOPT_WRITEFUNCTION, WriteMemoryCallback);
+  if(S3fsCurl::is_verbose){
+    curl_easy_setopt(hCurl, CURLOPT_DEBUGFUNCTION, S3fsCurl::CurlDebugBodyInFunc);     // replace debug function
```
Comment (gaul) [inline src/curl.cpp:3584]: Should this call the following as above:

```cpp
curl_easy_setopt(hCurl, CURLOPT_VERBOSE, true);
```

Repeated below.

File: src/curl.h [hunk 2]
```diff
@@ -279,6 +279,9 @@ class S3fsCurl
     static sse_type_t       ssetype;
     static bool             is_content_md5;
     static bool             is_verbose;
+//TEST
+    static bool             is_dump_body;
+//TEST
```
[the remark this answers is not in the document]
Comment (ggtakec) [reply src/curl.h:284]: Sorry my fault.
It was leaking to delete comments for testing.
Rebase and upload again.

File: src/curl.h [hunk 3]
```diff
@@ -386,6 +389,11 @@ class S3fsCurl
     static bool AddUserAgent(CURL* hCurl);
 
     static int CurlDebugFunc(CURL* hcurl, curl_infotype type, char* data, size_t size, void* userptr);
+//TEST
```
[the remark this answers is not in the document]
Comment (ggtakec) [reply src/curl.h:392]: Same above.

File: src/curl.cpp (hunk 1 above)
Comment (ggtakec) [reply src/curl.cpp:3584]: CURLOPT_VERBOSE is set in ResetHandle() called from CreateCurlHandle method.
Here we are replacing(Resetting) the default CurlDebugFunc handler set by ResetHandle() with CurlDebugBodyInFunc.
So this is no problem.

**Pull request merged**